internal/monitors/conviva/conviva.go

```diff
@@ -0,0 +1,254 @@
+	MetricConfigs               []*MetricConfig              `yaml:"metricConfigs"`
+	filterNameById              map[string]map[string]string
+	dimensionIdByAccountAndName map[string]map[string]string
+	metriclensMetricNames       map[string][]string
+}
+
+type MetricConfig struct {
+	Account             string   `yaml:"account"`
+	Metric              string   `yaml:"metric" default:"quality_metriclens"`
+	Filters             []string `yaml:"filters"`
+	Dimensions          []string `yaml:"dimensions"`
+	accountId           string
+	filterIds           []string
+	metriclensFilterIds []string
+}
+
+// Monitor for conviva metrics
+type Monitor struct {
+	Output types.Output
+	cancel func()
+	ctx     context.Context
+	client *http.Client
+	timeout time.Duration
+}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+func (m *Monitor) Configure(conf *Config) error {
+	logger.Debugf("configuration object before additional auto-configuration:\n%+v\n", conf)
+	m.timeout = time.Duration(conf.TimeoutSeconds) * time.Second
+	m.client = &http.Client{
+		Timeout: m.timeout,
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: 100,
+		},
+	}
+	setConfigFields(m, conf)
+	maxNumOfGoroutines := len(conf.MetricConfigs) * 100
+	if maxNumOfGoroutines > 500 {
+		maxNumOfGoroutines = 500
+	}
+	maxNumOfGoroutinesChan := make(chan struct{}, maxNumOfGoroutines)
+	utils.RunOnInterval(m.ctx, func() {
+		for _, metricConfig := range conf.MetricConfigs {
+			if strings.Contains(metricConfig.Metric, "metriclens") {
+				getSendMetriclensMetrics(maxNumOfGoroutinesChan, m, metricConfig, conf)
+			} else {
+				getSendMetrics(maxNumOfGoroutinesChan, m, metricConfig, conf)
+			}
+		}
+	}, time.Duration(conf.IntervalSeconds) * time.Second)
+	return nil
+}
+//TODO: implement functionality for simple series and label series
+func getSendMetrics(maxNumOfGoroutinesChan chan struct{}, m *Monitor, metricConfig *MetricConfig, conf *Config) {
+	ctx, cancel := context.WithTimeout(m.ctx, time.Duration(conf.IntervalSeconds) * time.Second)
+	select {
+	case <- ctx.Done():
+		cancel()
+		logger.Error(ctx.Err())
+		return
+	case maxNumOfGoroutinesChan <- struct{}{}:
+		url := "https://api.conviva.com/insights/2.4/metrics.json?metrics=" + metricConfig.Metric +
+			"&filter_ids=" + strings.Join(metricConfig.filterIds, ",")
+		go func(m *Monitor, conf *Config, url string) {
+			defer func() {<- maxNumOfGoroutinesChan}()
+			res, err := get(m, conf, url)
+			if err != nil {
+				logger.Error(err)
+				return
+			}
+			timestamps := res[metricConfig.Metric].(map[string]interface{})["timestamps"].([]interface{})
+			timeSeriesByFilterId := res[metricConfig.Metric].(map[string]interface{})["filters"].(map[string]interface{})
+			dps := make([]*datapoint.Datapoint, 0)
+			for filterId, timeSeries := range timeSeriesByFilterId {
+				for i, metricValue := range timeSeries.([]interface{}) {
```

Comment:
We probably don't want historical data, only the latest metrics when they are queried.